src/extra/Markdown/index.tsx

```diff
@@ -138,11 +139,111 @@ type components = {
 	[element: string]: (props: any) => JSX.Element | React.ReactElement | Element;
 };
 
+interface Decorator {
+	match: RegExp;
+	component: string;
+}
+
+interface DecoratorPluginOptions {
+	decorators: Decorator[];
+}
+
+const decoratorPlugin = (options: DecoratorPluginOptions) => {
+	options = options || { decorators: [] };
+
+	if (!options.decorators || !options.decorators.length) {
+		return;
+	}
+
+	const targets: Array<{ decorator: Decorator; node: any }> = [];
+
+	const visitor = (node: any) => {
+		const matchedDecorator = options.decorators.find((decorator) => {
+			return (
+				node.children &&
+				node.children.length === 1 &&
+				node.children[0].type === 'text' &&
+				decorator.match.test(node.children[0].value) &&
+				node.children[0]
+			);
+		});
+
+		if (matchedDecorator) {
+			targets.push({
+				decorator: matchedDecorator,
+				node,
+			});
+		}
+	};
+
+	const modify = () => {
+		targets.forEach((target) => {
+			const node = target.node;
+			const match = target.decorator.match;
+
+			const text = node.children[0].value;
+			const children = [];
+
+			match.lastIndex = 0;
+			do {
+				const lastIndex = match.lastIndex;
+				const result = match.exec(text);
+
+				if (result) {
+					children.push(
+						{
+							type: 'element',
+							tagName: 'span',
+							children: [
+								{
+									type: 'text',
+									value: text.substring(lastIndex, result.index),
+								},
+							],
+						},
+						{
+							type: 'element',
+							tagName: 'mention',
```

Comment:
It would be better if it is a `span` with a class to style it.

Reply:
Thanks for reminding that, I forgot to change tagName to `target.decorator.component`. This is dynamic,  so pass whatever tag you want.